internal/plugins/helm/v1/scaffolds/templates/manager/config.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Modifications copyright 2020 The Operator-SDK Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/pkg/model/file"
+)
+
+var _ file.Template = &Config{}
+
+// Config scaffolds yaml config for the manager.
+type Config struct {
+	file.TemplateMixin
+	file.DomainMixin
+	file.RepositoryMixin
+
+	// Image is controller manager image name
+	Image string
+}
+
+// SetTemplateDefaults implements input.Template
+func (f *Config) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = filepath.Join("config", "manager", "manager.yaml")
+	}
+
+	f.TemplateBody = configTemplate
+
+	if f.Repo == "" {
+		dir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("error to get the current path: %v", err)
+		}
+		f.Repo = filepath.Base(dir)
+	}
```

Comment:
If `.Repo` is set to `github.com/myorg/my-operator`, we'll set the OPERATOR_NAME environment variable to that value, which IIRC is used for the leader election configmap name.

I think it is invalid to have slashes in configmap names.

We could change this by removing the `RepositoryMixin`, adding an `OperatorName` field, and then running a validation on it to verify it is not empty and is valid for a configmap name. Then we'd have to populate this field from the init scaffolder (which should have the project directory already).

Reply:
Hi @joelanford, 

1. How the repo would be `github.com/myorg/my-operator` for helm/ansible? 

Note that for GO the repo is == dir project name which is used as the name of the manager and etc. 
For GO is just possible to have github.com/myorg/my-operator because it has a flag --repo. 

2. regards `I think it is invalid to have slashes in configmap names.`

For KB and SDK CLI features the OperatorName == dir project which came from the repo field. I mean, If the repo has `github.com/myorg/my-operator` for example, all places will use `my-operator` only. So, why not follow up the same? 

3. Regards: 

> We could change this by removing the RepositoryMixin, adding an OperatorName field, and then running a validation on it to verify it is not empty and is valid for a configmap name. Then we'd have to populate this field from the init scaffolder (which should have the project directory already).

IMO: 

All validations are done when we call init to gen the project which ensures that the dir name is passing the K8S rules and can be used to create all. It is done before start to the scaffold. See that if it fails here then files were already scaffolded which is not a nice experience for users since they will need to understand that and clean up the dir to re-try again. Also, shows that we are adding unnecessary complexity. See that, I do not think that we should do as it is now either. IMO we should come back and keep it == KB implementation. 

**Suggested Solution:**

I think we should keep the repo, set the value when we do the init and keep the same implementation == across all types and KB (as it was done before). It would keep the simplicity and let it easier to maintain. Also, I understand that it was suggested by @estroz too in the PR https://github.com/operator-framework/operator-sdk/pull/3341#issuecomment-654710285 which was just required to be done because we are not using the repo. 

WDYT? Could we agree in came back to have the repo as it is done for Go and KB?

Reply:
Just to register. We spoke and I got @joelanford suggestion working on it.